### DISCUSSION TITLE
snippets/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#50] [Announcements]
# snippets/

30 Seconds of Java Inspired by 30 seconds of code, this is a collection of reusable tested copy-pasteable Java 17 compatible code snippets that you can understand in 30 seconds or less. If you're interested in contributing to this library, please see the instructions.

https://java-design-patterns.com/snippets/

**Comment by malek-glitch** [comment]
this is a great documentation

**Comment by SrBedrock** [comment]
Great for those learning Java! Thanks!

**Comment by Suiyueemperor** [comment]
bug: CycleSort
if (arr[i] < n && arr[i] != arr[correctpos])->if (arr[i] <= n && arr[i] != arr[correctpos])
eg. [5 4 2 1 3] , when i = 0, arr[i] < n is false.

**Comment by praveenkumargit1996** [comment]
Great documentation . Great start for all developers to start with this.
